test/SurplusBeneficiary/SurplusBeneficiary.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { smock } from "@defi-wonderland/smock"
+import chai, { expect } from "chai"
+import { solidity } from "ethereum-waffle"
+import { parseUnits } from "ethers/lib/utils"
+import { ethers, waffle } from "hardhat"
+import { FeeDistributor, SurplusBeneficiary, TestERC20, VePERP } from "../../typechain"
+import { getLatestTimestamp } from "../shared/utilities"
+
+chai.use(solidity)
+
+describe("SurplusBeneficiary spec", () => {
+    const [admin, alice] = waffle.provider.getWallets()
+    let vePERP: VePERP
+    let feeDistributor: FeeDistributor
+    let surplusBeneficiary: SurplusBeneficiary
+    let testPERP: TestERC20
+    let testUSDC: TestERC20
+    let treasury: TestERC20
+    const daoPercentage = 0.42e6 // 42%
+    const DAY = 86400
+    const WEEK = DAY * 7
+    const MONTH = DAY * 30
+    const YEAR = DAY * 365
+
+    beforeEach(async () => {
+        const testERC20Factory = await ethers.getContractFactory("TestERC20")
+        testPERP = await testERC20Factory.deploy()
+        await testPERP.__TestERC20_init("PERP", "PERP", 18)
+
+        testUSDC = await testERC20Factory.deploy()
+        await testUSDC.__TestERC20_init("USDC", "USDC", 6)
+
+        // use erc20 contract as a treasury contract
+        treasury = await testERC20Factory.deploy()
+
+        const vePERPFactory = await ethers.getContractFactory("vePERP")
+        vePERP = (await vePERPFactory.deploy(testPERP.address, "vePERP", "vePERP", "v1")) as VePERP
+
+        const feeDistributorFactory = await ethers.getContractFactory("FeeDistributor")
+        feeDistributor = (await feeDistributorFactory.deploy(
```

Comment:
i think we can just mock a feeDistributor instead of deploying an instance for unit test

Reply:
change to use feeDistributor mock contract in spec.ts